source/event_loop.c

```diff
@@ -311,6 +327,50 @@ void aws_event_loop_clean_up_base(struct aws_event_loop *event_loop) {
     aws_hash_table_clean_up(&event_loop->local_data);
 }
 
+void aws_event_loop_register_tick_start(struct aws_event_loop *event_loop) {
+    aws_high_res_clock_get_ticks(&event_loop->latest_tick_start);
+}
+
+void aws_event_loop_register_tick_end(struct aws_event_loop *event_loop) {
+    /* increment the timestamp diff counter (this should always be called from the same thread), the concurrency
+     * work happens during the flush. */
+    uint64_t end_tick = 0;
+    aws_high_res_clock_get_ticks(&end_tick);
+
+    aws_add_size_saturating(event_loop->current_tick_latency_sum, end_tick - event_loop->latest_tick_start);
+    event_loop->latest_tick_start = 0;
+
+    size_t next_flush_time_secs = aws_atomic_load_int(&event_loop->next_flush_time);
```

Comment:
trivial: we could avoid this atomic_load by having a separate non-atomic copy of this variable

Reply:
it's not something i'm incredibly concerned about. We've already processed the tick, and there's about 4 kernel calls on both sides of anything calling this function.